numpy/core/_dtype_ctypes.py

```diff
@@ -33,17 +33,28 @@ def _from_ctypes_array(t):
 
 
 def _from_ctypes_structure(t):
-    # TODO: gh-10533, gh-10532
-    fields = []
-    for item in t._fields_:
-        if len(item) > 2:
-            raise TypeError(
+    # TODO: gh-10533
+    if hasattr(t,"_pack_"):
+        fields = {}
```

Comment:
The `dtype({name: type})` constructor isn't safe to use, as it fails if one of the fields is called "formats".

I'd recommend using:
```python
np.dtype(dict(
    formats=[...],  # dtypes
    offsets=[...],  # integers
    names=[...],    # strings
    itemsize=ctypes.sizeof(t)
))
```
Which doesn't suffer from this fault.

Reply:
Ok, I changed the code. I did not use ctypes.sizeof(t) as it does not take into consideration the _pack_ attribute it has (at least in my tests). 
For example: 
`class Foo(ctypes.Structure):
    _fields_ = [('one', ctypes.c_uint8), ('two', ctypes.c_uint32)]
    _pack_ = 6`

results in a Structure that has size 8 (while I expect size 12 - as it has 2 items in 2 6 long regions). I am a bit unsure if correct size is 12(items_nr * pack) or 10 (last offset + last item size). Or is the correct behavior to accept only `pack <= max(type size)` and use default pack when the `pack > max(type size)`?
This is how I understood the alignment for this case (2 items, sizes 1 and 4) is that the first item starts on 0, the second one starts on 6. To put it short, the items are placed at offsets divisible by pack. This is not happening for the above example, two has offset 4 in the ctypes structure (as PyCharm debugger shows it).

Reply:
I think this line reveals the secret:

https://github.com/python/cpython/blob/e42b705188271da108de42b55d9344642170aa2b/Modules/_ctypes/cfield.c#L154-L155

So yes, pack is only used if it's smaller than the default alignment.

Reply:
Note: maybe I could add another (or several) test that would better force out errors. Something like a more complex structure: 

```
Class Foo(ctypes.Structure):
    _fields_ = [('1one', ctypes.c_uint8), ('2two', ctypes.c_uint16), ('3three', ctypes.c_uint8),('4four', ctypes.c_uint16),('5five', ctypes.c_uint32), ('6six', ctypes.c_uint32)]
    _pack_ = 3 
```

I am not sure if we want several tests for similar functionalities. Please let me know.